packages/rocketchat-lib/server/functions/updateMessage.js

```diff
@@ -28,16 +28,21 @@ RocketChat.updateMessage = function(message, user, originalMessage) {
 		RocketChat.models.Messages.cloneAndSaveAsHistoryById(message._id);
 	}
 
-	message.editedAt = new Date();
-	message.editedBy = {
-		_id: user._id,
-		username: user.username,
-	};
 
-	const urls = message.msg.match(/([A-Za-z]{3,9}):\/\/([-;:&=\+\$,\w]+@{1})?([-A-Za-z0-9\.]+)+:?(\d+)?((\/[-\+=!:~%\/\.@\,\w]*)?\??([-\+=&!:;%@\/\.\,\w]+)?(?:#([^\s\)]+))?)?/g) || [];
-	message.urls = urls.map((url) => ({ url }));
 
-	message = RocketChat.callbacks.run('beforeSaveMessage', message);
+	if (originalMessage.msg !== message.msg) {
```

Comment:
should we return before having the message [processed by Apps](https://github.com/RocketChat/Rocket.Chat/pull/13053/files#diff-03e4fa2f14f1ee1fbf0fd78345355273L18)?

Reply:
Do you mean message processing by Apps should  happen when the messages are not equal ?

Reply:
Also, I am not sure what `Apps` does here.

Reply:
`Apps` can change the message content or prevent a message from being sent.

so my question was: should we skip everything if the edited message didn't change? or should `Apps` still process messages that were "edited" but still have the same content?

Reply:
@rodrigok @d-gubert @graywolf336 what you guys think on above?

Reply:
I'd say that we could skip the whole `RocketChat.updateMessage` method when the message's content doesn't change,   not executing `Apps`, before/after hooks, or message history -> https://github.com/RocketChat/Rocket.Chat/blob/235d0dbb73a69f5498b2b982b3bcee519c66d99e/packages/rocketchat-lib/server/functions/updateMessage.js#L27

Reply:
cool then sounds like this is good?

Reply:
Yeah I'll change it to the skipping the whole ``` RocketChat.updateMessage ``` method.